service/history/api/describeworkflow/api.go

```diff
@@ -333,18 +333,9 @@ func buildCallbackInfo(
 	callback callbacks.Callback,
 	outboundQueueCBPool *circuitbreakerpool.OutboundQueueCircuitBreakerPool,
 ) (*workflowpb.CallbackInfo, error) {
-	destination := ""
-	cbSpec := &commonpb.Callback{}
-	switch variant := callback.Callback.Variant.(type) {
-	case *persistencespb.Callback_Nexus_:
-		cbSpec.Variant = &commonpb.Callback_Nexus_{
-			Nexus: &commonpb.Callback_Nexus{
-				Url:    variant.Nexus.GetUrl(),
-				Header: variant.Nexus.GetHeader(),
-			},
-		}
-		destination = variant.Nexus.GetUrl()
-	default:
+	cbSpec, err := workflow.PersistenceCallbackToAPICallback(callback.Callback)
+	if err != nil || cbSpec.GetNexus() == nil {
+		// Errors only happen for non-nexus callbacks.
```

Comment:
Hesitant to make such assumptions in the code. If the function intends to mark something as an ignorable error it should probably just return nil or have a specific error type that you can filter out.